src/ray/gcs/gcs_client/service_based_gcs_client.cc

```diff
@@ -39,28 +39,32 @@ Status ServiceBasedGcsClient::Connect(boost::asio::io_service &io_service) {
   gcs_pub_sub_.reset(new GcsPubSub(redis_gcs_client_->GetRedisClient()));
 
   // Get gcs service address.
-  auto get_server_address = [this]() {
+  get_server_address_func_ = [this]() {
     std::pair<std::string, int> address;
     GetGcsServerAddressFromRedis(redis_gcs_client_->primary_context()->sync_context(),
```

Comment:
Now we are calling `GetGcsServerAddressFromRedis` periodically. It shouldn't retry and sleep, otherwise it will block the event loop

Reply:
GetGcsServerAddressFromRedis will be retried only when gcs client is initialized, because we need to wait for GCS server to start.